rascal2/widgets/project.py

```diff
@@ -0,0 +1,278 @@
+    """Combobox that saves previous value"""
+
+    def __init__(self, combo_box: QtWidgets.QComboBox, selected_value: str, previous_value: str):
+        """
+        Initialize combobox.
+
+        Parameters
+        ----------
+        combo_box: QtWidgets.QComboBox
+                The combobox whose value needs to be updated.
+        selected_value: str
+                The new value of the combobox.
+        previous_value: str
+                The previous value of the combobox.
+        """
+        super().__init__()
+        self.combo_box = combo_box
+        self.selected_value = selected_value
+        self.previous_value = previous_value
+
+    def undo(self) -> None:
+        """Method to undo the last change"""
+        self.combo_box.setCurrentText(self.previous_value)
+
+
+class ProjectWidget(QtWidgets.QWidget):
+    """
+    The Project MDI Widget
+    """
+
+    def __init__(self, parent):
+        """
+        Initialize widget.
+
+        Parameters
+        ----------
+        parent: MainWindowView
+                An instance of the MainWindowView
+        """
+        super().__init__(parent)
+        self.parent = parent
+        self.presenter = self.parent.presenter
+        self.model = self.parent.presenter.model
+        self.undo_stack = QtGui.QUndoStack(self)
+
+        self.create_project_view()
+        self.create_edit_view()
+        self.handle_domains_tab()
+        self.update_model_project_view()
+
+        self.stacked_widget = QtWidgets.QStackedWidget()
+        self.stacked_widget.addWidget(self.project_widget)
+        self.stacked_widget.addWidget(self.edit_project_widget)
+
+        layout = QtWidgets.QVBoxLayout()
+        layout.addWidget(self.stacked_widget)
+        self.setLayout(layout)
+
+    def update_model_project_view(self) -> None:
+        """Updates the project view when a new project is created or values are updated in the model."""
+        if self.model.project:
+            self.project_calculation = self.model.project.calculation
+            self.project_model = self.model.project.model
+            self.project_geometry = self.model.project.geometry
+
+            self.inter_calculation = self.model.project.calculation
+            self.inter_model = self.model.project.model
+            self.inter_geometry = self.model.project.geometry
+        else:
+            self.project_calculation = ""
+            self.project_model = ""
+            self.project_geometry = ""
+            self.inter_calculation = ""
+            self.inter_model = ""
+            self.inter_geometry = ""
+
+        self.calculation_type.setText(self.project_calculation)
+        self.model_type.setText(self.project_model)
+        self.geometry_type.setText(self.project_geometry)
+
+        self.calculation_combobox.setCurrentText(self.project_calculation)
+        self.model_combobox.setCurrentText(self.project_model)
+        self.geometry_combobox.setCurrentText(self.project_geometry)
```

Comment:
you won't need to handle the case where `self.model.project` doesn't exist if you remove `self.update_model_project_view()` from `ProjectWidget.__init__` and put it in `MainWindowView.setup_mdi`: see there that `setup_mdi` is used for widget setup that can't be done properly until the project is open

Reply:
I did something similar in the first commit and was told that I can't have signals if they are being used in just one place in the code in a previous review.

Reply:
<img width="786" alt="image" src="https://github.com/user-attachments/assets/af6ad3e5-1d1f-4a47-8d3d-299a5cd75731">

I refactored it because you have asked me not have single use signals. It would be a single use signal and this was my way of not making it a single use signal. Honestly, I just want to wrap up this project. I will make the changes based on what I understand and it can be refactored later by someone.